src/Tmds.Ssh/Curve25519KeyExchange.cs

```diff
@@ -0,0 +1,215 @@
+// This file is part of Tmds.Ssh which is released under MIT.
+// See file LICENSE for full license details.
+
+using System.Buffers;
+using System.Security.Cryptography;
+using System.Numerics;
+using Microsoft.Extensions.Logging;
+using Org.BouncyCastle.Crypto.Generators;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Security;
+using Org.BouncyCastle.Crypto.Prng;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Agreement;
+
+namespace Tmds.Ssh;
+
+// Curve25519 Key Exchange: https://datatracker.ietf.org/doc/html/rfc8731
+class Curve25519KeyExchange : IKeyExchangeAlgorithm
+{
+    private readonly HashAlgorithmName _hashAlgorithmName;
+
+    public Curve25519KeyExchange(HashAlgorithmName hashAlgorithmName)
```

Comment:
nit: remote the constructor argument, as we're never setting it to anything other than `HashAlgorithmName.SHA256`.

Reply:
Do you want me rename the file as `Curve25519Sha256KeyExchange.cs`?

Reply:
Renamed in https://github.com/tmds/Tmds.Ssh/pull/311